src/ui.rs

```diff
@@ -2524,7 +2524,22 @@ impl Component for FileView<'_> {
         if self.is_expanded() {
             let x = x + 2;
             let mut section_y = y + file_view_header_rect.height.unwrap_isize();
-            for section_view in section_views {
+            let mut expanded_sections = HashSet::new();
+            for (i, view) in section_views.iter().enumerate() {
+                if view.is_expanded() && view.section.is_editable() {
+                    expanded_sections.insert(i);
+                }
+            }
```

Comment:
The only reason I didn't do this is because it ends up being more lines of code. But agreed that scoping the mutability is generally a good idea. I'll change it.